main.tf

```diff
@@ -48,7 +53,7 @@ resource "azurerm_public_ip" "primary" {
 }
 
 resource "azurerm_network_interface" "dynamic" {
-  name                = local.name
+  name                = local.virtual_machine_name

```

Comment:
@jhisc in my mind I had thought to keep the resource name the same and simply edit the hostname with the computer_name option that is present for both linux and windows VMs. But this gets the same result and keeps the name the same in the portal